### PR TITLE
Add prow jobs to ensure backend code quality

### LINKED ISSUE #5116: Add prow jobs for verifying kubermatic API code
**User story**
As a developer, I would like to have some PR hooks that would verify my kubermatic-api code before it is merged to the upstream.

**Acceptance criteria**
- (`pre-kubermatic-verify`) copy verification scripts from https://github.com/kubermatic/kubermatic repository and run prow job to verify swagger, code-gen, import order etc. for open PRs
- (`pre-kubermatic-lint`) run go linter for kubermatic-api code
- unittests

### PR DESCRIPTION
**Pull request by pkprzekwas**
**What this PR does / why we need it**:
As we moved API code to this repository we need to copy appropriate prow pipelines to validate and test the code.

This PR adds following prow jobs:
- unit tests for API's go code
- go linter
- verify: 
  - import order
  - go mod deps
  - license compatibility
  - boilerplate
  - shell scripts formatting
  - spellcheck

**NOTE:** spellcheck doesn't check any JS files for now.

**Which issue(s) this PR fixes**:
Fixes #5116 

**What type of PR is this?**
/kind chore

**Special notes for your reviewer:**
Number of changed files in this PR is mainly a result of running import ordering tool.

**Does this PR introduce a user-facing change? Then add your Release Note here**:
```release-note
NONE
```

**Documentation**:
```documentation
NONE
```

### CODE REVIEW

File: .golangci.apis.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Copyright 2022 The Kubermatic Kubernetes Platform contributors.
```
Comment (ahmedwaleedmalik) [inline .golangci.apis.yml:1]: We don't need this.

File: .gimps.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Copyright 2021 The Kubermatic Kubernetes Platform contributors.
```
Comment (ahmedwaleedmalik) [inline .gimps.yaml:1]:
```suggestion
# Copyright 2022 The Kubermatic Kubernetes Platform contributors.
```

File: .prow/api.yaml [hunk 3]
```diff
@@ -51,3 +51,62 @@ presubmits:
...
               memory: 6Gi
+  - name: pre-dashboard-api-unit
+    run_if_changed: "^(cmd|hack|pkg)/"
+    decorate: true
+    clone_uri: "ssh://git@github.com/kubermatic/dashboard.git"
+    labels:
+      preset-goproxy: "true"
+    spec:
+      containers:
+        - image: quay.io/kubermatic/build:go-1.19-node-18-2
```
Comment (ahmedwaleedmalik) [inline .prow/api.yaml:62]: We recently updated to Go 1.19.3 for the KKP API so you'll have to switch these CI jobs to `quay.io/kubermatic/build:go-1.19-node-18-4`

**Review by ahmedwaleedmalik — CHANGES_REQUESTED** [review]

File: .golangci.apis.yml (hunk 1 above)
Comment (pkprzekwas) [reply .golangci.apis.yml:1]: Oh, good catch. It's CRD only, right?

Comment (ahmedwaleedmalik) [reply .golangci.apis.yml]: Yup that's correct.

**Comment by ahmedwaleedmalik** [comment]
> pre-dashboard-test-full-e2e-ce

`cat: hack/ci/testdata/oauth_values.yaml: No such file or directory

**Comment by pkprzekwas** [comment]
> We can ignore pre-dashboard-test-e2e and pre-dashboard-test-e2e-ce but might need a fix for the following
> 
> > pre-dashboard-test-full-e2e-ce pre-dashboard-test-full-e2e-ce
> 
> `cat: hack/ci/testdata/oauth_values.yaml: No such file or directory`

Failing test seems to be a result of https://github.com/kubermatic/kubermatic/pull/11258/files

**Review by ahmedwaleedmalik — APPROVED** [review]
/approve 🚀

**Comment by ahmedwaleedmalik** [comment]
Closed and reopened the PR since it was stuck in the merge pool.

**Pull request merged**